### DISCUSSION TITLE
Can't load application from file

### DISCUSSION THREAD

**Discussion by daleif** [mojolicious/mojo#2109] [Q&A]
I've just moved to a new work PC. So I now develop under perl 5.34 and Mojolicious 9.34

I use morbo under development and I've noticed that if there was a mistake in my code I no longer get a stack trace of the problem

I only get 

```
Can't load application from file "path to script": path to script did not return a true value at path/Mojo/Server.pm line 59.
```
this is really annoying and makes debugging rather hard.

What changes am I missing here coming from perl 5.26.1 and Mojolicious 9.31

**Comment by kraih** [comment]
That's been fixed in 9.34.

**Reply by daleif** [reply]
Wasn't that something about perl 5.38? This is 5.34. And I'm already running 9.34.

**Reply by kraih** [reply]
Then i'm afraid i have no idea what's going on.

**Reply by daleif** [reply]
I'll see if I can make a lite app with the same issues.

**Reply by daleif** [reply]
Test file, perl 5.34, Mojolicious 9.34 (example inspired by the example in #2094)

```
use v5.34;
use Mojolicious::Lite -signatures;
$test;
app->start;
```
Just gives the same error while running it with `morbo` (`Can't load application from file "path to script": script did not return a true value at ...`)

Adjusted for perl 5.26 on my old PC:

```
use v5.26;
use Mojolicious::Lite -signatures;
$test;
app->start;
```
here returns morbo gives

```
Can't load application from file "path to script": Global symbol "$test" requires explicit package name (did you forget to declare "my $test"?) at "script" line 3.
Compilation failed in require at (eval 93) line 1.
```
This was Mojolicious 9.33

I might be needing a module somewhere.

**Reply by daleif** [reply]
I cannot be the only one running on Ubuntu 22.04

**Reply by daleif** [reply]
The problem seems to be Mojolicous 9.34. The example works as expected with 9.33 on my old PC, and locally installing 9.34 and running the same test program shows this problem.

Can this discussion be moved to Issues or should I manually log it there.

**Reply by daleif** [reply]
Note that it works as expected if I revert the change in `Mojo/Server.pm` back to using `require` instead of `do`.

**Reply by kraih** [reply]
If this is a regression make sure to reference #2097.

**Reply by daleif** [reply]
Added as #2110

**Reply by llarochelle** [reply]
I'm running Ubuntu 22.04 with Perl 5.34 and Mojolicious 9.34. I'm encoutering this bug. I tried using "require" instead of "do" in Server.pm on line 57 and get an issue with route method not being found.

luc@ubuntu:~/dev/Perl/Mojolicious-BoilerPlate$ morbo script/boilerplate
Web application available at http://127.0.0.1:3000
Can't load application from file "/home/luc/dev/Perl/Mojolicious-BoilerPlate/script/boilerplate": /home/luc/dev/Perl/Mojolicious-BoilerPlate/script/boilerplate did not return a true value at /usr/local/share/perl/5.34.0/Mojo/Server.pm line 59
